src/spineml/simulator/logOutput.cc

```diff
@@ -274,16 +233,73 @@ void SpineMLSimulator::LogOutput::AnalogueNetwork::record(double, unsigned int t
     }
 }
 
+//----------------------------------------------------------------------------
+// SpineMLSimulator::LogOutput::AnalogueNetwork
+//----------------------------------------------------------------------------
+SpineMLSimulator::LogOutput::AnalogueNetwork::AnalogueNetwork(const pugi::xml_node &node, double dt,
+                                                              const std::string &port, unsigned int popSize,
+                                                              const filesystem::path &logPath,
+                                                              const ModelProperty::Base *modelProperty)
+    : AnalogueExternal(node, dt, port, popSize, logPath, modelProperty)
+{
+    // Check size determined by indices/population size matches attribute
+    const unsigned int size = getIndices().empty() ? popSize : (unsigned int)getIndices().size();
+    //assert(size == node.attribute("size").as_uint());
+
+    // Allocate output buffer
+    m_OutputBuffer.resize(size);
+
+    // Read connection stats
+    const std::string connectionName = node.attribute("name").value();
+    const std::string hostname = node.attribute("host").value();
+    const int tcpPort = node.attribute("tcp_port").as_int();
```

Comment:
I believe port numbers are always ``uint16_t``. You might get a warning about a narrowing conversion, depending on how fussy your compiler is

Reply:
I guess, for the sake of fussy compilers, we could add a cast here https://github.com/genn-team/genn/blob/spineml_simulator_library/src/spineml/simulator/networkClient.cc#L60?

Reply:
Yeah, that would work, although port numbers should always be ``uint16_t``, so you could just use that from the outset.

Reply:
So, in general I favour passing native-sized types around, but I have changed it to parse and subsequently pass the port around as ``unsigned int``  and cast to ``uint16_t`` at the ``htons`` call to hopefully appease fussy compilers

Reply:
:+1: